projects/components/table/src/subcomponents/table-pagination.component.ts

```diff
@@ -86,18 +86,20 @@ export class PsTablePaginationComponent implements OnInit, OnDestroy {
       .subscribe(pageEvent => this.page.emit(pageEvent));
   }
 
-  public ngOnInit() {
-    this.dataSource._internalDetectChanges.pipe(takeUntil(this.ngUnsubscribe$)).subscribe(() => {
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.pageSize || changes.dataLength) {
       this.pages = [];
-      for (let i = 0; i < this.dataSource.pages; i++) {
+      const pageCount = Math.ceil(this.dataLength / this.pageSize);
+      for (let i = 0; i < pageCount; i++) {

```

Comment:
you could use 
`this.pages = Array.from({ length: pageCount }, (_, k) => k + 1);`